### ISSUE TITLE
Bug when running axiom-init, Error: open /tmp/raman18.sh: no such file or directory

### ISSUE THREAD

**Issue by pry0cc** [pry0cc/axiom#40]
```~$ axiom-init --restore=pry0cc/lazy
Using restore backup ‘pry0cc/lazy’
Initializing ‘raman18’
Error: open /tmp/raman18.sh: no such file or directory
Initialized! Instance accessible at ‘’!
To connect, run ‘axiom-ssh raman18’
Waiting 65 seconds before restore…
Restoring pry0cc-lazy to raman18 at ’ ’
ssh: Could not resolve hostname : Name or service not known
rsync: connection unexpectedly closed (0 bytes received so far) [sender]
rsync error: unexplained error (code 255) at io.c(235) [sender=3.1.2]```

This occurs because there is no valid image detected, you need to run `axiom-build` and receive a successful build for your currently selected region.

This is because axiom now checks for axiom-$region as apposed to axiom* in your images, running `axiom-build` will however resolve this.

**Comment by daddyhack** [comment]
>This is because axiom now checks for axiom-$region as apposed to axiom* in your images, running `axiom-build` will however resolve this.

No matter what i try I cannot get past this bug. I have built and rebuilt multiple times and still can no longer get "axiom-init" to run successfully. It was working beautifully in the past.

**Comment by pry0cc** [comment]
I have an idea.

Stand by. I think this will fix it for you. Sorry about this, open source is hard :P

**Comment by b00ls0ck3t** [comment]
i'm facing the same issues described by daddyhack 

![image](https://user-images.githubusercontent.com/19557578/87345131-df707980-c54f-11ea-9e97-aea82767aa77.png)

**Comment by pry0cc** [comment]
Can you provide more details, namely, operating system, bash version and doctl version.

```
uname -a
cat /etc/os-release
bash --version
doctl version
axiom-update
```

**Comment by natgriffiths** [comment]
I'm in the same position as daddyhack i.e. can no longer to get axiom-init to run successfully. These are my details:
Linux Private 4.15.0-111-generic #112-Ubuntu SMP Thu Jul 9 20:32:34 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

NAME="Linux Mint"
VERSION="19.1 (Tessa)"
ID=linuxmint
ID_LIKE=ubuntu
PRETTY_NAME="Linux Mint 19.1"
VERSION_ID="19.1"
HOME_URL="https://www.linuxmint.com/"
SUPPORT_URL="https://forums.ubuntu.com/"
BUG_REPORT_URL="http://linuxmint-troubleshooting-guide.readthedocs.io/en/latest/"
PRIVACY_POLICY_URL="https://www.linuxmint.com/"
VERSION_CODENAME=tessa
UBUNTU_CODENAME=bionic

GNU bash, version 4.4.20(1)-release (x86_64-pc-linux-gnu)
Copyright © 2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

doctl version 1.45.1-release
Git commit hash: 1041a0e5

axiom-update
Already up-to-date.

**Comment by daddyhack** [comment]
-------------------------------------------------------------------
> Can you provide more details, namely, operating system, bash version and doctl version.
> 
> ```
> uname -a
> cat /etc/os-release
> bash --version
> doctl version
> axiom-update
> ```

**uname -a:**
Linux testbox 5.4.0-40-generic #44-Ubuntu SMP Tue Jun 23 00:01:04 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

**cat /etc/os-release**
NAME="Ubuntu"
VERSION="20.04 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal

**bash --version**
GNU bash, version 5.0.16(1)-release (x86_64-pc-linux-gnu)

** doctl version**
doctl version 1.45.1-release
Git commit hash: 1041a0e5

**axiom-update**
remote: Enumerating objects: 11, done.
remote: Counting objects: 100% (11/11), done.
remote: Compressing objects: 100% (7/7), done.
remote: Total 11 (delta 4), reused 11 (delta 4), pack-reused 0
Unpacking objects: 100% (11/11), 2.45 KiB | 1.22 MiB/s, done.
From https://github.com/pry0cc/axiom
   49057b8..ff6012d  master     -> origin/master
Updating 49057b8..ff6012d
Fast-forward
 interact/axiom-ls    | 20 +++++++++++++++++---
 interact/axiom-proxy | 13 +++++++++++++
 2 files changed, 30 insertions(+), 3 deletions(-)
 create mode 100755 interact/axiom-proxy

-------------------

**Comment by xlaruen** [comment]
This problem happens when doctl is installed via snap since it's using strict confinement, which means doctl has no access to files on your filesystem (except the whitelisted directories configured in their snapcraft.yaml). One of possible workarounds is to remove doctl from snap and install it by other means, for example use their latest pre-built binary from https://github.com/digitalocean/doctl/tags

**Comment by daddyhack** [comment]
@xlaruen Thank you! Removing doctl via snap and reinstalling using the pre-built binary worked. I can once again run "axiom-init" successfully.

**Comment by natgriffiths** [comment]
@xlaruen thanks from me also. I can run axiom-init successfully again as well. Cheers mate.

**Comment by pry0cc** [comment]
Oh wow!

That makes a lot of sense! I did not think about that. I've been using both MacOS and Arch, I had no idea snap confined the filesystem like that!

We'll have to update the axiom-configure script then to pull the binaries for Ubuntu users, this new functionality with the --user-data-file is going to become the basis for a bit of more tech that I'm planning down the road. Right now we use it for expiration, every box has an expiration of about 10000 years, and then we can set that expiration to be sooner than default, and that's how we added the --expire feature.

Thanks for solving this one, appreciate this everybody!

pry0cc

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐
